### PR TITLE
docs: Add link to rustdocs for `master` to README

### PR DESCRIPTION
**Pull request by umgefahren**
## Description

I added a link to the master docs to the [README](README.md).

## Links to any relevant issues

Following #3126.

## Open Questions

I added a warning to the libp2p module documentation. This has to be removed before a release. Is this a good solution?

## Change checklist

- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -18,6 +18,9 @@
 // FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
 // DEALINGS IN THE SOFTWARE.
 
+// Remove this note before publishing
+//! # ⚠️ WARNING You are viewing the master docs
+//!
```
Comment (thomaseizinger) [inline src/lib.rs]: Pretty sure we are going to forget about this. I don't think we need to explicitly point this out. I'd expect users who navigate to https://libp2p.github.io/rust-libp2p/libp2p/ to understand that they are viewing the master docs and not of a released version.

**Review by thomaseizinger — CHANGES_REQUESTED** [review]

File: src/lib.rs (hunk 1 above)
Comment (umgefahren) [reply src/lib.rs]: Ok, will remove it then.

File: src/lib.rs [hunk 2]
```diff
@@ -18,8 +18,6 @@
 // FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
 // DEALINGS IN THE SOFTWARE.
 
-//! libp2p is a modular peer-to-peer networking framework.
```
Comment (jxs) [inline src/lib.rs]: was this also supposed to be removed?

Comment (umgefahren) [reply src/lib.rs]: Resolved in latest commit. Sorry for doing so many small mistakes.

**Review by thomaseizinger — APPROVED** [review]

File: src/lib.rs (hunk 2 above)
Comment (jxs) [reply src/lib.rs]: No worries Hannes, it happens a lot to me :sweat_smile:

Comment (umgefahren) [reply src/lib.rs]: That calms me :)

**Review by elenaf9 — DISMISSED** [review]

**Comment by thomaseizinger** [comment]
Adding `send-it`. Once #3213 is sorted (merged and the patch version is released), you can merge this by just updating it to latest master.

**Review by thomaseizinger — DISMISSED** [review]

**Review by thomaseizinger — DISMISSED** [review]

**Review by thomaseizinger — APPROVED** [review]

**Pull request merged**